core/src/main/java/com/datastax/oss/driver/internal/core/context/StartupOptionsBuilder.java

```diff
@@ -119,6 +122,12 @@ public Map<String, String> build() {
     if (applicationVersion != null) {
       builder.put(APPLICATION_VERSION_KEY, applicationVersion);
     }
+    if (applicationLocalDc == null) {
+      applicationLocalDc = localDc(config);
```

Comment:
_config_ here is [just the default profile](https://github.com/apache/cassandra-java-driver/blob/4.19.0/core/src/main/java/com/datastax/oss/driver/internal/core/context/StartupOptionsBuilder.java#L95) but any profile can define a local datacenter... which in turn means we need to think about how execution profiles interact with this functionality.

I'm going to raise this point on the JIRA ticket; I think it warrants further discussion.

Reply:
[Raised](https://issues.apache.org/jira/browse/CASSJAVA-92?focusedCommentId=17944601&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17944601)